### PR TITLE
Add behat tests for customer service

### LINKED ISSUE #16592: Migrate "Customer Service > Customer Service" dashboard page
Part of Symfony migration project
Type: page

## Specs

https://github.com/PrestaShop/prestashop-specs/pull/187

## Design

https://www.figma.com/file/axasD3Kyg52okWHqvWPN1I/Customer-Service?node-id=12%3A7862

## Improvements

- [ ] #16783 - Add the possibility to attach files, access predefined messages and resize the messages panel
- [ ] #19197 - Fix the bulk action

## Status

Todo:
- [ ] Add a modal like on product list (see below)
- [x] Migrate links to the page (see below), see PR #28634
- [x] Grid
- [ ] 2 forms
- [ ] Multistore header: [Specs](https://github.com/PrestaShop/prestashop-specs/blob/7e820f30347fe69ccd3db5a99ab02ef6f7bbdd54/back-office/multistoregeneralspecs.md#multistore-header-only-from-178)
- [ ] #19378 - Checkboxes for multistore
- [ ] #19330 - Specific settings dropdown for multistore

## Warnings

A lot of links in the BO target this page, pay attention to them.

- [ ] [Add a modal like on product list (Catalog > Products) to confirm or cancel delete in bulk action](https://github.com/PrestaShop/PrestaShop/issues/14462)
- [ ] [Add a modal to confirm or cancel delete action for one row](https://github.com/PrestaShop/PrestaShop/issues/17082)

#### Controller / routing
- [ ] All controller actions are migrated and protected with `@AdminSecurity` annotations when eligible and `@DemoRestricted` annotations when eligible
- [ ] All links targeting this page are updated: either using _legacy_link or updating the code
- [ ] The new SF routes have `_legacy_link` and `_legacy_parameters` provided and explained
- [ ] Controller catches Domain Exception and translate them nicely
- [ ] Add/Edit controllers must catch global Exception in order to handle ModuleErrorException
#### Template
- [ ] Twig templates must contain relevant blocks to allow partial extension by modules
- [ ] Twig forms must have `form_rest()` and `form_errors()`
- [ ] JS being used on the page can be used as enabled extensions and use a PageMap for selectors
- [ ] Use `@PrestaShop` instead of `PrestaShopBundle` for template root path aliasing
##### Conventions
- [ ] The new controller actions comply with our naming convention
- [ ] The new SF routes names and paths comply with our naming convention
- [ ] The new Twig template names comply actions with our naming convention
- [ ] Created files (controller, JS assets, CSS assets and twig views) must be in the right folder following the convention
- [ ] CSS being added must follow theme conventions

#### Core layer
- [ ] Commands and Queries only use native types but internally use VOs
- [ ] All Handlers must have an Interface
##### Conventions
- [ ] The created SF service names comply with our naming convention
- [ ] The created files must follow the folder convention (Domain, Core, Adapter...)

#### Polishing
- [ ] The legacy controller and templates files are gone
- [ ] If PR includes a new concept/mechanism, documentation must be written
- [ ] JS and CSS assets must be compiled and up-to-date
- [ ] License headers must be up-to-date
- [ ] New hooks have been added to the hooks list and the SQL updates
- [ ] Performance must be correct: page loading time, filtering/sorting/searching response time, actions response time

#### Testing
- [ ] Commands and Queries are tested with Behat tests
- [ ] Eligible Core classes are covered with unit tests*
- [ ] Controller is covered with survival test
- [ ] (soon) Controller is covered with E2E tests using puppeeter
- [ ] If we break E2E tests because IDs change, we must update the E2E tests

*Eligible = easy to isolate and unit tests provide values (for example testing getters/setters is useless)

### PR DESCRIPTION
**Pull request by margud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | customer service migration. Added behat tests
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Fixed ticket?     | Fixes #16592 
| How to test?      | Run command: ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s customer_service --tags customer-management

### CODE REVIEW

**Review by jolelievre — DISMISSED** [review]
Thanks @margud

**Review by zuk3975 — DISMISSED** [review]

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerServiceFeatureContext.php [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
```
Comment (nicosomb) [inline tests/Integration/Behaviour/Features/Context/Domain/CustomerServiceFeatureContext.php:20]: The URL is now https://devdocs.prestashop-project.org/

**Review by nicosomb — COMMENTED** [review]
juste a little comment

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerServiceFeatureContext.php (hunk 1 above)
Comment (jolelievre) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerServiceFeatureContext.php:20]: Agreed, but NTH modification not sure it should block the PR We have to uniformize this one the whole project anyway

**Review by jolelievre — APPROVED** [review]

**Review by zuk3975 — APPROVED** [review]

**Comment by zuk3975** [comment]
No QA needed as it only introduces some behat test scenarios.
Thanks @margud :tada:

**Pull request merged**